Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.h

```diff
@@ -336,6 +336,14 @@ class GD_CORE_API EventsCodeGenerator {
     return projectScopedContainers;
   }
 
+  /**
+   * @brief The code generator push variable containers in the same instance
+   * to avoid to pass an instance by parameter.

```

Comment:
This is not super clear

```suggestion
   * @brief Give access to the project scoped containers as code generation might
   * push and pop variable containers (for local variables).
   * This could be passed as a parameter recursively in code generation, but this requires
   * heavy refactoring. Instead, we use this single instance.
```